qcelemental/tests/test_molecule.py

```diff
@@ -617,3 +630,60 @@ def test_molecule_connectivity():
     connectivity[0][0] = -1
     with pytest.raises(ValueError):
         mol = Molecule(**data, connectivity=connectivity)
+
+
+def test_orient_nomasses():
+    """
+    Masses must be auto generated on the fly
+    """
+
+    mol = Molecule(symbols=["He", "He"], geometry=[0, 0, -2, 0, 0, 2], orient=True, validated=True)
+
+    assert mol.__dict__["masses_"] is None
+    assert compare_values([[2, 0, 0], [-2, 0, 0]], mol.geometry)
+
+
+@pytest.mark.parametrize(
+    "mol_string, extra_keys",
+    [
+        ("He 0 0 0", None),
+        ("He 0 0 0\n--\nHe 0 0 5", {"fragments", "fragment_charges", "fragment_multiplicities"}),
+        ("He 0 0 0\n--\n@He 0 0 5", {"fragments", "fragment_charges", "fragment_multiplicities", "real"}),
+        ("He4 0 0 0", {"atom_labels"}),
+        ("He@3.14 0 0 0", {"masses", "mass_numbers"}),
+    ],
+)
+def test_sparse_molecule_fields(mol_string, extra_keys):
+
+    expected_keys = {
+        "schema_name",
+        "schema_version",
+        "validated",
+        "symbols",
+        "geometry",
+        "name",
+        "molecular_charge",
+        "molecular_multiplicity",
+        "fix_com",
+        "fix_orientation",
+        "provenance",
+    }
+    mol = Molecule.from_data(mol_string)
+
+    if extra_keys is not None:
+        expected_keys |= extra_keys
+
+    diff_keys = mol.dict().keys() ^ expected_keys
+    assert len(diff_keys) == 0, f"Diff Keys {diff_keys}"
+
+
+def test_sparse_molecule_connectivity():
+    """
+    A bit of a weird test, but because we set connectivity it should carry through.
```

Comment:
you recognize that the same doesn't hold for https://github.com/MolSSI/QCElemental/pull/191/files#diff-2025ec5a9d08fc78a8e5239d9c715ef8L250-L251?

Reply:
The link doesn't seem to work, where is this pointing to? But yes, I realize that connectivity is a special case. I wanted to pull it out for display so that we can modify in the future.

Reply:
the latter two "fragment_" args in 
```def test_from_data_kwargs():
    mol = Molecule.from_data(
        """
        O 0 0 0
        H 0 1.5 0
        H 0 0 1.5
        """,
        molecular_charge=1,
        molecular_multiplicity=2,
        fragment_charges=[1],
        fragment_multiplicities=[2],
```

ok, good in connectivity as special case.